tests/kernel/install_kotd.pm

```diff
@@ -102,15 +103,19 @@ END
 
 sub run {
     my $self = shift;
-    $self->wait_boot;
 
+    $self->wait_boot;
     $self->select_serial_terminal;
 
-    my $url = get_required_var("INSTALL_KOTD");
-    if ($url !~ /^https?:\/\//) {
-        my $arch = get_required_var("ARCH");
-        $url = "http://download.suse.de/ibs/Devel:/Kernel:/$url/standard/$arch/";
+    my $baseurl = "http://download.suse.de/ibs/Devel:";
+    my $version = "SLE" . get_required_var("VERSION");
+    if (is_opensuse) {
+        $baseurl = "http://download.opensuse.org/repositories";
+        $version = "HEAD";
     }
+    my $default_url = "$baseurl/Kernel:/$version/standard";
+    my $url         = get_var("KOTD_REPO", $default_url);
+    $url .= "/" . get_required_var("ARCH") . "/";
 
     if (grub_version() == 1) {
         my $file = download_kernel($url, 'kernel-default-base');
```

Comment:
Just looking on the logic how is current installation of KOTD done, it is so complex that it deserves rewrite. I would rather follow first guide from https://en.opensuse.org/openSUSE:Kernel_of_the_day . Although we are testing kernel, we really don't need manual download and manual install in such way. Of course it is not in the scope of this PR.

Reply:
Agree. Feel free to improve it more. This was supposed to be a quick enhancement to address broken variables and openSUSE, more should be planned in sprint.